docs/content/direct/combined-status.md

```diff
@@ -194,4 +193,52 @@ select:
 
 ## Special case for 1 WEC
 
-This proposal refines the meaning of `BindingPolicySpec.WantSingletonReportedState` to request the special case when it applies.
+When a workload object is distributed from a WDS to exactly one WEC,
+the reported state from that WEC can be returned into the copy of the
+workload object in the WDS. The design of most kinds of Kubernetes API
+object implicitly assumes that the object exists and has its defined
+effect in only one cluster. That is why it makes sense to return the
+reported state from the WEC to the WDS only when the object goes to
+exactly one WEC.
+
+As mentioned above, currently KubeStellar equates "reported state"
+with the `.status` section of the API object.
+
+The user has to specifically request this last step of `.status`
+propagation. This is done in an optional boolean field, named
+`wantSingletonReportedState`, in a `DownsyncPolicyClause` in a
+`BindingPolicy`. This is one of the three kinds of downsync
+modulations that a policy clause can associate with the matching
+workload objects. In a `Binding`, this same optional boolean field
+appears in `NamespaceScopeDownsyncClause` and
+`ClusterScopeDownsyncClause`. If multiple policy clauses in a
+`BindingPolicy` match a given workload object, the settings for
+`.wantSingletonReportedState` are combined by OR to get the one
+boolean value that appears with the reference to the object in the
+corresponding `Binding`. If multiple `Binding` objects in one WDS
+reference a given workload object, there is another level of
+multiplicity to consider. Read on.
+
+For a given workload object and WDS, we say that "singleton status
+return is requested" if and only if there exists at least one
+BindingPolicy or Binding that has `wantSingletonReportedState==true`
+in a clause that matches/references the workload object.
+
+The "qualified WEC set" of a given workload object in a given WDS is
+the set of WECs that are associated with that workload object by at
+least one BindingPolicy or Binding that has
+`wantSingletonReportedState==true` in a clause that matches/references
+the workload object.
+
+For a given workload object in a given WDS, while singleton status
+return is requested, KubeStellar maintains a label on the object whose
+name (key) is `kubestellar.io/executing-count` and whose value is a
+string representation of the size of the qualified WEC set of that
+object.  While singleton status return is _not_ requested, KubeStellar
+suppresses the existence of a label with that name (key).  While
+singleton status return is requested _and_ the size of the qualified
+WEC set is 1, KubeStellar propagates the object's `.status` from that
```

Comment:
The case of single status return is requested and the size of the qualified WEC set is bigger than 1 is not addressed. What happens in this case?

Reply:
Huh? The last sentence of this paragraph covers that case.

Reply:
Sorry I see it now, was not visible earlier for some reason.

You mean that if the system is operating well in a case of:
- A user's BP requests singleton status and selects a single WEC
- A different BP is deployed and the total of selected WECs per relevant objects from the above step increases
Then status would first be propagated and finally get dropped when the system reconciles the 2nd point?

Reply:
Yes

Reply:
Sorry we're having tough times here. Why is this preferable over allowing the first to "own" the status delivery and rejecting the rest that follow?

Reply:
That would raise additional issues. Perhaps there should be a documented definition of which WEC's status is being returned to the WDS (I am not sure what you mean by "first" in our concurrent and dynamic world). For the sake of not thrashing, the implementation should fix on one WEC and stay latched onto that WEC as long as it is in the qualified set, but switch to another when that WEC drops out of the qualified set. The user will want a way of knowing _which_ WEC's status is being returned to the WDS. I think that it is simpler to just return the status to the WDS only when the qualified WEC set size is 1.

Reply:
It would probably make a lot more sense with "WEC selection RBAC". 
I think for starters we shouldn't manage racing BPs, and that was not the point in the previous message. My concern is unnecessarily de-propagating status due to same-user errors (having another BP propagate a resource to another WEC) or a different user in another BP.

The challenge in the behavior + API you're proposing is that if one user marks a resource for singleton status, then no one can ship the same resource to another WEC, for any purpose.

Though it is fair to completely shift the responsibility onto the users. I just wonder if we can achieve a more robust behavior without heavy changes to the proposal.

Reply:
> Another concern in the behavior + API you're proposing is that if one user marks a resource for singleton status, then no one can ship the same resource to another WEC, for any purpose.

No. Note that this PR and #2408 differ from #2406 in this matter. This is why the "qualified WEC set" consists only of WECs that the object is bound to with `wantSingletonReportedState=true`.

Because there is only one status section in a workload object in the WEC, I do not think we have any real chance of supporting independent uncoordinated users of the same workload object in the same WDS.

Reply:
We can not merge just the API change, the PR that gets merged needs to change the implementation too. It should be a relatively small change.

Reply:
Point is lets move forwards with this direction. I agree it is the simplest, I still think that at the very least some form of feedback should help users understand their incoordination/bad use.

Reply:
Agreed on moving forward.
Agree on the desirability of feedback. We had a design, up hrough https://github.com/kubestellar/kubestellar/blob/release-0.24.0/api/control/v1alpha1/types.go#L119-L130 , but that declaration has been deleted since the implementation was deleted when we switched to KSLight. There is a similar design in https://github.com/MikeSpreitzer/kcp-edge-mc/blob/select-to-wds/api/control/v1alpha1/types.go#L427-L428 . There is also a design somewhere for giving feedback in the `.status` of each Binding[Policy].